RUFAS/routines/EEE/emissions.py

```diff
@@ -509,25 +506,114 @@ def _calculate_emissions_by_field(
+    def _apply_fertilizer_to_next_crop(
+        self,
+        fertilizer_application: dict[str, Any],
+        sorted_crops: list[dict[str, Any]],
+        fertilizer_application_date: datetime.date,
+    ) -> bool:
+        """
+        Applies fertilizer to the next available crop after harvest and before the next planting.
+        Returns True if fertilizer was applied, False otherwise.
+        """
+        for i, crop in enumerate(sorted_crops):
+            crop_harvest_date = Time.convert_year_jday_to_date(crop["harvest_year"], crop["harvest_day"])
+            if i + 1 < len(sorted_crops):
+                next_crop = sorted_crops[i + 1]
+                next_crop_planting_date = Time.convert_year_jday_to_date(
+                    next_crop["planting_year"], next_crop["planting_day"]
+                )
+                if crop_harvest_date < fertilizer_application_date < next_crop_planting_date:
+                    next_crop["nitrogen_fertilizer_used"] += fertilizer_application["nitrogen"]
+                    next_crop["nitrogen_fertilizer_embedded_CO2_emissions"] = (
+                        fertilizer_application["nitrogen"] * EMBEDDED_NITROGEN_FERTILIZER_EMISSIONS_FACTOR
+                    )
+                    next_crop["phosphorus_fertilizer_used"] += fertilizer_application["phosphorus"]
+                    next_crop["phosphorus_fertilizer_embedded_CO2_emissions"] = (
+                        fertilizer_application["phosphorus"] * EMBEDDED_PHOSPHORUS_FERTILIZER_EMISSIONS_FACTOR
+                    )
+                    next_crop["potassium_fertilizer_used"] += fertilizer_application["potassium"]
+                    next_crop["potassium_fertilizer_embedded_CO2_emissions"] = (
+                        fertilizer_application["potassium"] * EMBEDDED_POTASSIUM_FERTILIZER_EMISSIONS_FACTOR
+                    )
+                    return True
+        return False
+
+    def _extract_applied_crops(
+        self, sorted_crops: list[dict[str, Any]], fertilizer_application_date: date
+    ) -> list[dict[str, Any]]:
+        """Extracts a list of crops that had fertilizer applied to them between their planting and harvesting dates.
+
+        Parameters
+        ----------
+        sorted_crops : list[dict[str, Any]]
+            The list of crops in the field sorted by planting date.
+        fertilizer_application_date : date
+            The date of the fertilizer applciation.
+
+        Returns
+        -------
+        list
+            A list of the crops whose planting and harvesting dates encompass the fertilizer application date.
+        """
+        applied_crops = []
+
+        for crop in sorted_crops:
+            crop_planting_date = Time.convert_year_jday_to_date(crop["planting_year"], crop["planting_day"])
+            crop_harvest_date = Time.convert_year_jday_to_date(crop["harvest_year"], crop["harvest_day"])
+
+            if crop_planting_date <= fertilizer_application_date <= crop_harvest_date:
```

Comment:
```suggestion
            if crop_planting_date <= fertilizer_application_date < crop_harvest_date:
```
If a fertilizer application occurred on the same day that a crop was harvested, I think it would be more reasonable to assume that the fertilizer application should not be assigned to that crop and instead to the next. @krb78 what do you think?